src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp

```diff
@@ -693,6 +693,11 @@ void InterpreterMacroAssembler::remove_activation(
     // testing if reserved zone needs to be re-enabled
     Label no_reserved_zone_enabling;
 
+    // check if already enabled - if so no re-enabling needed
+    ldr(rscratch1, Address(rthread, JavaThread::stack_guard_state_offset()));
+    cmp(rscratch1, (u1)StackOverflow::stack_guard_enabled);
```

Comment:
Not ldrw + cmpw?

Reply:
I've no idea, I used the code below as the pattern here. @theRealAph reviewed this.

Reply:
ldrw + cmpw does seem more correct, plus an asset that sizeof _stack_guard_state == 4.  ldr+cmp is only going to work for little-endian, and only as long as the alignment padding between _stack_guard_state and _stack_overflow_limit is all zeroes.

Reply:
Changed